build/commands/scripts/sync.js

```diff
@@ -41,40 +50,209 @@ const maybeInstallDepotTools = (options = config.defaultOptions) => {
+      custom_deps: {
+        'src/third_party/WebKit/LayoutTests': '%None%',
+        'src/chrome_frame/tools/test/reference_build/chrome': '%None%',
+        'src/chrome_frame/tools/test/reference_build/chrome_win': '%None%',
+        'src/chrome/tools/test/reference_build/chrome': '%None%',
+        'src/chrome/tools/test/reference_build/chrome_linux': '%None%',
+        'src/chrome/tools/test/reference_build/chrome_mac': '%None%',
+        'src/chrome/tools/test/reference_build/chrome_win': '%None%'
+      },
+      custom_vars: {
+        'checkout_pgo_profiles': config.isBraveReleaseBuild() ? '%True%' :
+                                                                '%False%'
+      }
+    },
+    {
+      managed: '%False%',
+      name: 'src/brave',
+      // We do not use gclient to manage brave-core, so this should not
+      // actually get used.
+      url: 'https://github.com/brave/brave-core.git'
+    }
+  ])
+
+  if (process.env.GIT_CACHE_PATH) {
+    out += toGClientConfigItem('cache_dir', process.env.GIT_CACHE_PATH)
+  }
+  if (config.targetOS) {
+    out += toGClientConfigItem('target_os', [config.targetOS], false)
+  }
+
+  fs.writeFileSync(config.defaultGClientFile, out)
+}
+
+function shouldUpdateChromium(latestSuccessfulSyncInfo, expectedSuccessfulSyncInfo) {
+  const chromiumRef = expectedSuccessfulSyncInfo.chromiumRef
+  const headSHA = util.runGit(config.srcDir, ['rev-parse', 'HEAD'], true)
+  const targetSHA = util.runGit(config.srcDir, ['rev-parse', chromiumRef], true)
+  const needsUpdate = targetSHA !== headSHA || (!headSHA && !targetSHA) ||
+      JSON.stringify(latestSuccessfulSyncInfo) !==
+          JSON.stringify(expectedSuccessfulSyncInfo)
+  if (needsUpdate) {
+    const currentRef = util.getGitReadableLocalRef(config.srcDir)
+    console.log(
+        `Chromium repo ${chalk.blue.bold('needs sync')}.\n  target is ${
+            chalk.italic(chromiumRef)} at commit ${
+            targetSHA || '[missing]'}\n  current commit is ${
+            chalk.italic(currentRef || '[unknown]')} at commit ${
+            chalk.inverse(
+                headSHA || '[missing]')}\n  latest successful sync is ${
+                  JSON.stringify(latestSuccessfulSyncInfo, null, 4)}`)
+  } else {
+    console.log(
+        chalk.green.bold(`Chromium repo does not need sync as it is already ${
+            chalk.italic(
+                chromiumRef)} at commit ${targetSHA || '[missing]'}.`))
+  }
+  return needsUpdate
+}
+
+function syncChromium(program) {
+  const requiredChromiumRef = config.getProjectRef('chrome')
+  let args = [
+    'sync', '--nohooks', '--reset', '--revision',
+    'src@' + requiredChromiumRef, '--with_tags',
+    '--with_branch_heads', '--upstream'
+  ];
+
+  const syncWithForce = program.init || program.force
+  if (syncWithForce) {
+    args.push('--force')
+  }
+
+  if (program.sync_chromium_and_delete_unused_deps) {
+    if (util.isGitExclusionExists(config.srcDir, 'brave/')) {
```

Comment:
what about unused deps in `src/brave`?

Reply:
I thought that they are still removed even if a dir is not managed by `gclient`, but I need to check that.